test/parallel/test-fs-watch-recursive.js

```diff
@@ -2,6 +2,8 @@
 
 const common = require('../common');
 
+if (common.isIBMi)
+  common.skip('IBMi does not support `fs.watch()`');
```

Comment:
Why? Shouldn't your changes make it so that it works everywhere now?

Reply:
The name of the test is `test/parallel/test-fs-watch-recursive.js`, I don't think it would create any false assumption. Keeping this line does give the false assumption that IBMi is not supported.

Reply:
`fs.watch()` is not implemented on IBM i (no underlying mechanism to do so AFAIK). So all tests that use `fs.watch()` should be skipped on IBM i.

Reply:
The same thing applies for AIX. `test-fs-watch-encoding.js` file includes a if statement:

```
if (common.isAIX)
  common.skip('folder watch capability is limited in AIX.');
```

Reply:
The difference is that IBM i doesn't support fs.watch at all while AIX does but with limitations.

Reply:
I'm thinking about adding support for `{ recursive: false }` and just enable it for non-windows and non-macOS OS, and just remove all of these skip checks. Wdyt?

Reply:
My main goal is to keep https://ci.nodejs.org/job/node-test-commit-ibmi/ at least unstable (yellow) and not to introduce new test failures.
We run that CI daily but not on pull requests because unfortunately it is very slow. Feel free to start runs of that job through Jenkins if you need to check anything on IBM i.